src/tls.rs

```diff
@@ -489,24 +498,28 @@ impl Handshake {
 
     pub fn peer_cert(&self) -> Option<Vec<u8>> {
         let peer_cert = unsafe {
-            let mut out: *mut libc::c_uchar = ptr::null_mut();
+            let chain = SSL_get0_peer_certificates(self.as_ptr());
+            if chain.is_null() {
```

Comment:
Rather than have all these `is_null()` checks, it's probably better to use `map_result_ptr()`, e.g.
```rust
             let chain = map_result_ptr(SSL_get0_peer_certificates(self.as_ptr())).ok()?;
```

Reply:
sure why not ....